replica/table.cc

```diff
@@ -368,8 +368,8 @@ future<std::vector<locked_cell>> table::lock_counter_cells(const mutation& m, db
 }
 
 std::vector<memtable*> table::active_memtables() {
-    return boost::copy_range<std::vector<memtable*>>(compaction_groups() | boost::adaptors::transformed([] (compaction_group& cg) {
-        return &cg.memtables()->active_memtable();
+    return boost::copy_range<std::vector<memtable*>>(compaction_groups() | boost::adaptors::transformed([] (compaction_group* cg) {
```

Comment:
This adds an allocation, in `compaction_groups()`.
We better get the size first to reserve capacity for the returned vector, then iterate through all storage/compaction groups, rather than returning a temporary vector to iterate over.

Please run `perf-simple-query --write` to compare performance / allocations impact of this change.

Reply:
I see that the function is currently not called on the hot read/write path, but it might be in the future.
But there is no reason to over-simplify it, leaving a time bomb behind.

Reply:
Yes, not a hot path today and I am planning to get rid of compaction_groups() in a near-future patch. replace it by for_each_active_memtable() is a good call, I think.